demosplan/DemosPlanCoreBundle/Command/Documentation/EventFinder.php

```diff
@@ -0,0 +1,340 @@
+<?php declare(strict_types=1);
+
+
+namespace demosplan\DemosPlanCoreBundle\Command\Documentation;
+
+
+use DemosEurope\DemosplanAddon\Exception\JsonException;
+use DemosEurope\DemosplanAddon\Utilities\Json;
+use demosplan\DemosPlanCoreBundle\Command\CoreCommand;
+use demosplan\DemosPlanCoreBundle\Exception\InvalidArgumentException;
+use demosplan\DemosPlanCoreBundle\Utilities\DemosPlanPath;
+use demosplan\DemosPlanCoreBundle\ValueObject\EventMatch;
+use demosplan\DemosPlanCoreBundle\ValueObject\UnnamedEventMatch;
+use PhpParser\Node\Stmt\Class_;
+use PhpParser\Node\Stmt\Namespace_;
+use PhpParser\NodeFinder;
+use PhpParser\ParserFactory;
+use PHPUnit\Util\Exception;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+use Webmozart\Assert\Assert;
+
+class EventFinder extends CoreCommand
+{
+    protected static $defaultName = 'dplan:documentation:generate:demos-event-list';
+    protected static $defaultDescription = '';
+
+    private const OPTION_START_PATHS = 'startPaths';
+    private const OPTION_PARENTS = 'parents';
+
+    private array $unnamedEventMatches = [];
+    private array $namedEventMatches = [];
+
+    /**
+     * The list of events, created by this command may be incomplete and/or contains false-positives,
+     * caused by limited options to identify relevant events as such, while event classes are not loaded.
+     * To do so, the name of the class will be used too, by looking for the term "event" as part of the name.
+     */
+    protected function configure(): void
+    {
+        $this->addOption(
+            self::OPTION_PARENTS,
```

Comment:
Extracting these to constants is very uncommon in the Symfony world. Especially since PhpStorm provides autocompletion for the strings in most usages anyway.

Reply:
I got no autocompletion provided (on usage `getOption()`), but even if I did, I would suggest to keep this in a constant, because these are used as keys.
What would be the alternative? just using the strings on `addOption()` and on `getOption()`?

Reply:
Using the strings is the only way I have ever seen in Symfony code and documentation, yes. Also, you may want to check your Symfony Integration Plugin in PHPStorm.